aws-predictions/src/main/java/com/amplifyframework/predictions/aws/service/AWSComprehendService.java

```diff
@@ -0,0 +1,338 @@
+import com.amazonaws.services.comprehend.model.DetectDominantLanguageRequest;
+import com.amazonaws.services.comprehend.model.DetectDominantLanguageResult;
+import com.amazonaws.services.comprehend.model.DetectEntitiesRequest;
+import com.amazonaws.services.comprehend.model.DetectEntitiesResult;
+import com.amazonaws.services.comprehend.model.DetectKeyPhrasesRequest;
+import com.amazonaws.services.comprehend.model.DetectKeyPhrasesResult;
+import com.amazonaws.services.comprehend.model.DetectSentimentRequest;
+import com.amazonaws.services.comprehend.model.DetectSentimentResult;
+import com.amazonaws.services.comprehend.model.DetectSyntaxRequest;
+import com.amazonaws.services.comprehend.model.DetectSyntaxResult;
+import com.amazonaws.services.comprehend.model.DominantLanguage;
+import com.amazonaws.services.comprehend.model.PartOfSpeechTag;
+import com.amazonaws.services.comprehend.model.SentimentScore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Predictions service for performing text interpretation.
+ */
+final class AWSComprehendService {
+
+    private final AmazonComprehendClient comprehend;
+    private final AWSInterpretConfiguration configuration;
+
+    AWSComprehendService(@Nullable AWSInterpretConfiguration configuration) {
+        this.comprehend = createComprehendClient();
+        this.configuration = configuration;
+    }
+
+    private AmazonComprehendClient createComprehendClient() {
+        AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance();
+        ClientConfiguration configuration = new ClientConfiguration();
+        configuration.setUserAgent(UserAgent.string());
+        return new AmazonComprehendClient(credentialsProvider, configuration);
+    }
+
+    void comprehend(
+            @NonNull String text,
+            @NonNull Consumer<InterpretResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        try {
+            // First obtain the dominant language to begin analysis
+            final Language dominantLanguage = fetchPredominantLanguage(text);
+            final LanguageType language = dominantLanguage.getValue();
+
+            // Actually analyze text in the context of dominant language
+            final Sentiment sentiment = fetchSentiment(text, language);
+            final List<KeyPhrase> keyPhrases = fetchKeyPhrases(text, language);
+            final List<Entity> entities = fetchEntities(text, language);
+            final List<Syntax> syntax = fetchSyntax(text, language);
+
+            onSuccess.accept(InterpretResult.builder()
+                    .language(dominantLanguage)
+                    .sentiment(sentiment)
+                    .keyPhrases(keyPhrases)
+                    .entities(entities)
+                    .syntax(syntax)
+                    .build());
+        } catch (PredictionsException exception) {
+            onError.accept(exception);
```

Comment:
Are we failing the whole operation if any of the above service fails? I think iOS sends partial result back to the user.

Reply:
discussed. JS also fails the entire operation so this will stay, but I can definitely see the merit in returning partial result. Perhaps we can do something similar to what we have in API category? `InterpretResult` can also contain error field, which will contain the list of service errors encountered. The error callback will only trigger on network error or validation error